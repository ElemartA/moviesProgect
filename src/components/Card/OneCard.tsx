import React, { FunctionComponent } from "react";
import { Card } from "react-bootstrap";
import s from "./OneCard.module.scss";

type TCardProps = {
  img: string;
  nameRu: string;
  ratingKinopoisk: number;
  year: string;
  onClick: () => void;
};

const OneCard: FunctionComponent<TCardProps> = ({
  nameRu,
  ratingKinopoisk,
  year,
  img,
  onClick,
}) => (
  <div>
    <Card
      className="bg-dark text-white"
      style={{ margin: 10, width: 270, boxShadow: `0px 1px 15px 2px black` }}
      onClick={onClick}
      data-testid="card"
    >
      <Card.Img
        src={img}
        alt="Card image"
        style={{ width: 268, height: 402 }}
      />
      <Card.ImgOverlay className={s.block}>
        <div className={s.block__title}>
          <Card.Title>{nameRu}</Card.Title>
          <Card.Text>Рейтинг: {ratingKinopoisk}</Card.Text>
          <Card.Text data-testid="year-id">Год выпуска: {year}</Card.Text>
        </div>
      </Card.ImgOverlay>
    </Card>
  </div>
);

export default OneCard;
