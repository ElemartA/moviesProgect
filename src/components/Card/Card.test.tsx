import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import OneCard from "./OneCard";

describe("Card", () => {
  it("renders Card component", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <OneCard
          img={""}
          nameRu={""}
          ratingKinopoisk={0}
          year={""}
          onClick={function (): void {
            throw new Error("Function not implemented.");
          }}
        />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
    screen.debug();
    expect(screen.getByTestId("card")).toBeInTheDocument();
    expect(screen.getByRole("img")).toBeInTheDocument();
  });
});
