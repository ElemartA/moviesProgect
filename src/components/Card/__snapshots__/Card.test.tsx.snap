// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders Card component 1`] = `
<DocumentFragment>
  <div>
    <div
      class="bg-dark text-white card"
      data-testid="card"
      style="margin: 10px; width: 270px; box-shadow: 0px 1px 15px 2px black;"
    >
      <img
        alt="Card image"
        class="card-img"
        src=""
        style="width: 268px; height: 402px;"
      />
      <div
        class="block card-img-overlay"
      >
        <div
          class="block__title"
        >
          <div
            class="card-title h5"
          />
          <p
            class="card-text"
          >
            Рейтинг: 0
          </p>
          <p
            class="card-text"
            data-testid="year-id"
          >
            Год выпуска: 
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
